CMakeLists.txt

```diff
@@ -140,18 +140,55 @@ if(NOT Boost_FOUND)
 endif()
 include_directories("${Boost_INCLUDE_DIR}")
 
-#add_executable(main main.cpp)
+add_executable(main 
+  main.cpp
+  graph.cpp
+  supernode.cpp
+  l0_sampling/sketch.cpp
+  l0_sampling/update.cpp
+  util.cpp
+)
+target_link_libraries(main PRIVATE xxHash::xxhash)
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+  target_compile_options(main PRIVATE -O3)
+elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+  target_compile_options(main PRIVATE /O2)
+endif()
+
 add_executable(l0sampling
   l0sampling.cpp
   l0_sampling/sketch.cpp
   l0_sampling/update.cpp
+  util.cpp
 )
 target_link_libraries(l0sampling PRIVATE xxHash::xxhash)
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+  target_compile_options(l0sampling PRIVATE -O3)
+elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+  target_compile_options(l0sampling PRIVATE /O2)
+endif()
```

Comment:
I'm not sure why we need different compiler options based on the compiler here?

Reply:
Different compilers have different syntax for the optimization flags. This bit of code is copied from the previous iteration, iirc